src/Icons/tests/Unit/IconRendererTest.php

```diff
@@ -159,6 +159,45 @@ public static function provideAttributesWithDefaultAttributesCases()
         ];
     }
 
+    /**
+     * @dataProvider provideAriaHiddenCases
+     *
+     * @param array<string, string> $attributes
+     */
+    public function testRenderIconWithAutoAriaHidden(array $attributes, string $expectedSvg): void
+    {
+        $registry = $this->createRegistry([
+            'foo' => '<path d="M0 0L12 12"/>',
+        ]);
+        $iconRenderer = new IconRenderer($registry, [], ['auto_aria_hidden' => true]);
+
+        $svg = $iconRenderer->renderIcon('foo', $attributes);
+        $this->assertSame($expectedSvg, $svg);
+    }
+
+    /**
+     * @return iterable<array{array<string, string>, string}>
+     */
+    public static function provideAriaHiddenCases(): iterable
+    {
+        yield 'no attributes' => [
+            [],
+            '<svg aria-hidden="true"><path d="M0 0L12 12"/></svg>',
+        ];
+        yield 'aria-hidden attribute' => [
+            ['aria-hidden' => 'true'],
```

Comment:
I'd add a case where it's forced to false and another attribute is passed to, e.g. `aria-label`